velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -47,12 +47,18 @@ ReaderBase::ReaderBase(
 }
 
 void ReaderBase::loadFileMetaData() {
-  bool preloadFile_ = fileLength_ <= filePreloadThreshold_;
+  preloadFile_ = fileLength_ <= filePreloadThreshold_ ||
+      fileLength_ <= directorySizeGuess_;
   uint64_t readSize =
       preloadFile_ ? fileLength_ : std::min(fileLength_, directorySizeGuess_);
 
-  auto stream = input_->read(
-      fileLength_ - readSize, readSize, dwio::common::LogType::FOOTER);
+  std::unique_ptr<dwio::common::SeekableInputStream> stream = nullptr;
+  if (preloadFile_) {
+    stream = input_->readFile(fileLength_, dwio::common::LogType::FOOTER);
```

Comment:
Could we use `input_->read(0, fileLength_)` instead of create a new method `readFile`?

Reply:
`readFile` uses enqueue/load to populate underlying buffers, which is different from what `read` do.

Reply:
`dwio::common::LogType::FILE` since we are reading the entire file?